components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -133,6 +133,15 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     return ContentFilteringPolicy::kAllow;
   }
 
+  // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+  // Doing so can break logins or other site functionality. 
+  if (!IsThirdParty(url, first_party_origin)) {
+    if (rules_->mode == mojom::ContentFilterMode::BLOCK_COOKIES) {
```

Comment:
For First-Party cookies, we want to block them too right? Am I doing this right? @darinwf @chung-lai

Reply:
I think the idea was to map `BLOCK_REQUEST` OR `BLOCK_COOKIES` to `kBlockCookies` instead.